### PR TITLE
chore: additional work on FFI documentation.

### PR DESCRIPTION
**Pull request by jsantell**
* Expand @\class and @\memberof Doxygen relationships amongst FFI functions.
* Refine some FFI docs for consistent language and adapted to Doxygen's "summary" and "detailed" views (via newlines).
* Fix up rustdoc lints.

### CODE REVIEW

File: rust/noosphere/src/ffi/headers.rs [hunk 1]
```diff
@@ -3,8 +3,18 @@ use safer_ffi::prelude::*;
 #[derive_ReprC(rename = "ns_headers")]
 #[repr(opaque)]
 /// @class ns_headers_t
+/// An opaque struct representing name/value headers.
 ///
-/// TBD
+/// Headers are used in ns_sphere_file_t to assign metadata, like content type.
```
Comment (jsantell) [inline rust/noosphere/src/ffi/headers.rs:8]: An example of "class" examples. As we only have one big example for now, unclear if having individual (untested??) examples is better. Can we run non-rust tests as a part of rustdocs? Hmm

File: rust/noosphere/src/ffi/noosphere.rs [hunk 2]
```diff
@@ -98,26 +91,29 @@ pub fn ns_initialize(
 }
 
 #[ffi_export]
-/// De-allocate a [NoosphereContext]. Note that this will also drop every
-/// [SphereContext] that remains active within the [NoosphereContext].
+/// @memberof ns_noosphere_t
+/// Deallocate a ns_noosphere_t instance.
+///
+/// Note that this will also deallocate every ns_sphere_t that remains
```
Comment (jsantell) [inline rust/noosphere/src/ffi/noosphere.rs:97]: Notably, this extra new line here creates the fully detailed function on the docs page with extended information. We could annotate other data e.g. @\param, @\return for more clarity as well, which would also create the "full" function def upon generation.

**Review by jsantell — COMMENTED** [review]
Some followups:
* Annotating safer_ffi's built-in wrapper types (`slice_boxed_char_ptr`). The generated rust comments from here are a little inscrutable.
* Refining some global functions into a phantom "class", as of now they're a bit hidden e.g. `ns_string_free()`, accessible only from the `noosphere.h` "file" "global functions".
* Investigate if we can test C functions in rustdocs, and other ways we can integrate examples in documentation
* Add links to landing page so that JS isn't required to navigate docs.

File: rust/noosphere/src/key/insecure.rs [hunk 3]
```diff
@@ -56,7 +56,7 @@ impl InsecureKeyStorage {
     /// but for now we aren't certain if we can depend on the ability to
     /// enumerate keys in general WebAuthn cases.
     ///
-    /// See: https://www.w3.org/TR/webauthn-3/#client-side-discoverable-credential
+    /// See: <https://www.w3.org/TR/webauthn-3/#client-side-discoverable-credential>
```
Comment (cdata) [inline rust/noosphere/src/key/insecure.rs:59]: How do the angle brackets help in this case?

File: rust/noosphere/src/ffi/headers.rs (hunk 1 above)
Comment (cdata) [reply rust/noosphere/src/ffi/headers.rs:8]: I would actually prefer not to have code examples in docs for the very reason that we can't easily keep them up to date. Better to have actual tests!

Comment (jsantell) [reply rust/noosphere/src/ffi/headers.rs:8]: agreed, examples running against CI way preferable at this point, will remove this

**Comment by jsantell** [comment]
@cdata updated & rebased

**Review by cdata — APPROVED** [review]
Awesome work!

**Pull request merged**